File: UI/user_ui/src/types/activities.ts
export type Activity = {
  id: number;
  activity_type: string;
  description: string;
  occurred_at?: Date;
  entity_id?: number;
  entity_type?: string;
};

export interface RecentActivitiesResponse {
  total: number;
  activities: Activity[];
  limit: number;
  offset: number;
}

File: api/pydantic_models/activities.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional
from datetime import datetime


class Activity(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    activity_type: str
    description: str
    occurred_at: Optional[datetime]
    entity_id: Optional[int]
    entity_type: str


class ActivitiesResponse(BaseModel):
    activities: List[Activity]
    total: int
    limit: int
    offset: int
